admin/src/pages/teams/figma-teams-connector.tsx

```diff
@@ -0,0 +1,176 @@
+import LoadingButton from '@atlaskit/button/loading-button';
+import Form, { ErrorMessage, Field, FormFooter } from '@atlaskit/form';
+import SectionMessage from '@atlaskit/section-message';
+import TextField from '@atlaskit/textfield';
+import { token } from '@atlaskit/tokens';
+import { css } from '@emotion/react';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useState } from 'react';
+
+import { ConnectTeamSuccessScreen } from './connect-teams-success-screen';
+
+import { connectTeam } from '../../api';
+import { ConnectBanner, FigmaPermissionsPopup, Page } from '../../components';
+import { parseTeamIdFromFigmaUrl } from '../../utils';
+
+type ConnectTeamProps = {
+	onClose?: () => void;
+	site: string;
+};
+
+export function FigmaTeamConnector({ onClose, site }: ConnectTeamProps) {
+	const queryClient = useQueryClient();
+	const [validationError, setValidationError] = useState<string | null>(null);
+	const connectTeamMutation = useMutation({
+		mutationFn: async (teamId: string) => {
+			return (await connectTeam(teamId)).data;
+		},
+		onSuccess: () => {
+			return queryClient.invalidateQueries({ queryKey: ['teams'] });
+		},
+	});
+
+	let content: React.ReactNode = null;
+	if (connectTeamMutation.isSuccess) {
+		content = (
+			<ConnectTeamSuccessScreen
+				onClose={onClose}
+				teamName={connectTeamMutation.data.teamName}
+			/>
+		);
+	} else {
+		content = (
+			<div
+				css={css`
```

Comment:
More of a general comment, what do you think about extractin styles as constants into `styled.tsx` files (or just outside the component body, top of the file) and passing them in rather than inlining? It might help with component readability, I found it quite tricky to scan this file at a glance (at least in GitHub)

Reply:
maybe switching to use tailwind is the solution here? In general I'm in favor of collocating the style definition w the markup since the 2 are intrinsically related. (eg. flex-grow: 1 only makes sense if its parent has display: flex).